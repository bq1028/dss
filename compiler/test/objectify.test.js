const { parse } = require('postcss')
const { objectify } = require('postcss-js')

const compile = css => objectify(parse(css))

describe('objetify css', () => {
  it('simple rule', () => {
    expect(
      compile(`
      .a { color: red }
    `)
    ).toMatchSnapshot()
  })

  it('pseudo class', () => {
    expect(
      compile(`
      .a { :hover { color: red } }
    `)
    ).toMatchSnapshot()
  })

  it('simple and pseudo class', () => {
    expect(
      compile(`
      .a { color: pink; :hover { color: red } }
    `)
    ).toMatchSnapshot()
  })

  it('media and pseudo', () => {
    expect(
      compile(`
      .a { @media (min-width: 30px) { :hover { color: red } } }
    `)
    ).toMatchSnapshot()
  })

  it('multiple', () => {
    expect(
      compile(`
      .a { color: red; }
      .a { :hover { color: blue; } }
      .a {
        @media screen and (min-width: 30px) {
          color: hotpink
        }
      }
    `)
    ).toMatchSnapshot()
  })

  it('state-combinator-class', () => {
    expect(
      compile(`
      .a { :hover > & { color: blue; } }
    `)
    ).toMatchSnapshot()
  })
})
