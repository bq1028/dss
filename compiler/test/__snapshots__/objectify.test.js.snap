// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objetify css media and pseudo 1`] = `
Object {
  ".a": Object {
    "@media (min-width: 30px)": Object {
      ":hover": Object {
        "color": "red",
      },
    },
  },
}
`;

exports[`objetify css multiple 1`] = `
Object {
  ".a": Object {
    ":hover": Object {
      "color": "blue",
    },
    "@media screen and (min-width: 30px)": Object {
      "color": "hotpink",
    },
    "color": "red",
  },
}
`;

exports[`objetify css pseudo class 1`] = `
Object {
  ".a": Object {
    ":hover": Object {
      "color": "red",
    },
  },
}
`;

exports[`objetify css simple and pseudo class 1`] = `
Object {
  ".a": Object {
    ":hover": Object {
      "color": "red",
    },
    "color": "pink",
  },
}
`;

exports[`objetify css simple rule 1`] = `
Object {
  ".a": Object {
    "color": "red",
  },
}
`;

exports[`objetify css state-combinator-class 1`] = `
Object {
  ".a": Object {
    ":hover > &": Object {
      "color": "blue",
    },
  },
}
`;
